code/src/components/CameraButton.js

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react';
+import { Provider, useDispatch } from 'react-redux';
+import { BarcodeScanner } from 'components/BarcodeScanner';
+import { products } from 'reducers/products';
+
+export const CameraButton = () => {
+  const dispatch = useDispatch();
+  const [showCamera, setShowCamera] = useState(false);
+  console.log(showCamera);
+
+  const onDetected = (code) => {
+    //return (dispatch) => {
+    //dispatch(ui.actions.setLoading(true));
+    fetch(`https://world.openfoodfacts.org/api/v0/product/${code}.json`)
+      .then((data) => data.json())
+      .then((json) => {
+        dispatch(products.actions.setProducts(json));
+        console.log(json);
+        //dispatch(ui.actions.setLoading(false));
+      });
+    //}
+  };
+
+  const handleBarCodeScanner = () => {
+    setShowCamera(!showCamera);
+  };
+
+  return (
+    <>
+      {showCamera ? (
+        <BarcodeScanner onDetected={onDetected} className='barcode-scanner' />
+      ) : (
+        ''
```

Comment:
There's no need for a ternary here if you're returning an empty string. You can use `&&` to just mount the component if `showCamera` is true:

```
{showCamera && <BarcodeScanner onDetected={onDetected} className='barcode-scanner' />}
```